group-coordinator/src/main/java/org/apache/kafka/coordinator/group/OffsetMetadataManager.java

```diff
@@ -668,6 +697,8 @@ public OffsetFetchResponseData.OffsetFetchResponseGroup fetchOffsets(
         OffsetFetchRequestData.OffsetFetchRequestGroup request,
         long lastCommittedOffset
     ) throws ApiException {
+        final boolean requireStable = lastCommittedOffset == Long.MAX_VALUE;
```

Comment:
is it the case that when we query stable offsets we always have the long max value for lastCommittedOffset? Or in other words, we can't query a specific offset when we require stable?

Reply:
Correct. When the require stable flag is set in the offset fetch request, the read operation reads with `MAX_VALUE` in order to read the latest state. If the flag is not set, the read op reads with the last committed offset.